superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -271,4 +270,19 @@ class VisualizeModal extends React.PureComponent {
 VisualizeModal.propTypes = propTypes;
 VisualizeModal.defaultProps = defaultProps;
 
-export default VisualizeModal;
+function mapStateToProps(state) {
+  return {
+    datasource: state.datasource,
+    errorMessage: state.errorMessage,
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    actions: bindActionCreators(actions, dispatch),
+  };
+}
+
+export { VisualizeModal };
+export default connect(mapStateToProps, mapDispatchToProps)(VisualizeModal);
```

Comment:
we aren't really using a consistent pattern around this yet. currently the following components in SQLLab are connected to redux/mapping state to props: 
App.jsx
DataPreviewModal.jsx
QueryAutoRefresh.jsx
SouthPane.jsx
TabbedSqlEditors.jsx

for this component, we could pass `actions`, `datasource`, and `errorMessage` through the component tree down from `TabbedSqlEditors` if we wanted to avoid connecting to the store here.